src/pages/Campaign/CampaignButtonWithOptions.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useRef, useState } from 'react'
+import { CampaignData, CampaignLeaderboard } from 'state/campaigns/actions'
+import { Trans } from '@lingui/macro'
+import { ChevronDown } from 'react-feather'
+import styled, { css } from 'styled-components'
+import { Button } from 'theme'
+import useTheme from 'hooks/useTheme'
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { OptionsContainer } from 'pages/TrueSight/styled'
+import { ChainId } from '@kyberswap/ks-sdk-core'
+import { NETWORKS_INFO } from 'constants/networks'
+import { Flex, Text } from 'rebass'
+import useMixpanel, { MIXPANEL_TYPE } from 'hooks/useMixpanel'
+import { useActiveWeb3React } from 'hooks'
+import { useSelector } from 'react-redux'
+import { AppState } from 'state'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { TransactionResponse } from '@ethersproject/providers'
+import useSendTransactionCallback from 'hooks/useSendTransactionCallback'
+import axios from 'axios'
+import { BigNumber } from '@ethersproject/bignumber'
+import { useActiveNetwork } from 'hooks/useActiveNetwork'
+
+export default function CampaignButtonWithOptions({
+  campaign,
+  disabled = false,
+  type,
+}: {
+  campaign: CampaignData | undefined
+  disabled?: boolean
+  type: 'enter_now' | 'claim_rewards'
+}) {
+  const theme = useTheme()
+  const [isShowNetworks, setIsShowNetworks] = useState(false)
+  const { changeNetwork } = useActiveNetwork()
+  const containerRef = useRef<HTMLButtonElement>(null)
+  useOnClickOutside(containerRef, () => setIsShowNetworks(false))
+  const { mixpanelHandler } = useMixpanel()
+
+  const chainIds: ChainId[] = campaign
+    ? campaign[type === 'enter_now' ? 'chainIds' : 'rewardChainIds'].split(',').map(item => +item)
```

Comment:
dont use `(item => +item)`. It's bad in semantic. Use `map(Number)`, it's better to understand and as shorter than this.

Reply:
Ok will update this.

Reply:
Thanks